vscode/microsoft-kiota/src/openApiTreeProvider.ts

```diff
@@ -3,41 +3,106 @@ import * as os from 'os';
 import * as path from 'path';
 import * as vscode from 'vscode';
 import * as rpc from 'vscode-jsonrpc/node';
-import { connectToKiota, KiotaGetManifestDetailsConfiguration, KiotaLogEntry, KiotaManifestResult, KiotaOpenApiNode, KiotaShowConfiguration, KiotaShowResult, LockFile } from './kiotaInterop';
+import * as crypto from 'crypto';
+import { 
+    ClientObjectProperties, 
+    ClientOrPluginProperties, 
+    connectToKiota, 
+    KiotaGetManifestDetailsConfiguration, 
+    KiotaLogEntry, 
+    KiotaManifestResult, 
+    KiotaOpenApiNode, 
+    KiotaShowConfiguration, 
+    KiotaShowResult, 
+    ConfigurationFile, 
+    PluginObjectProperties } from './kiotaInterop';
 import { ExtensionSettings } from './extensionSettings';
+import { treeViewId } from './constants';
+import { updateTreeViewIcons } from './util';
 
 export class OpenApiTreeProvider implements vscode.TreeDataProvider<OpenApiTreeNode> {
     private _onDidChangeTreeData: vscode.EventEmitter<OpenApiTreeNode | undefined | null | void> = new vscode.EventEmitter<OpenApiTreeNode | undefined | null | void>();
     readonly onDidChangeTreeData: vscode.Event<OpenApiTreeNode | undefined | null | void> = this._onDidChangeTreeData.event;
     private apiTitle?: string;
+    private initialStateHash: string = '';
     constructor(
         private readonly context: vscode.ExtensionContext,
-        private readonly settingsGetter:() => ExtensionSettings,
+        private readonly settingsGetter: () => ExtensionSettings,
         private _descriptionUrl?: string,
         public includeFilters: string[] = [],
         public excludeFilters: string[] = []) {
-        
+
     }
     private _lockFilePath?: string;
-    private _lockFile?: LockFile;
+    private _lockFile?: ConfigurationFile | Partial<ConfigurationFile> = {};
     public get isLockFileLoaded(): boolean {
         return !!this._lockFile;
     }
-    public async loadLockFile(path: string): Promise<void> {
-      this.closeDescription(false);
-      this._lockFilePath = path;
-      const lockFileData = await vscode.workspace.fs.readFile(vscode.Uri.file(path));
-      this._lockFile = JSON.parse(lockFileData.toString()) as LockFile;
-      if (this._lockFile?.descriptionLocation) {
-        this._descriptionUrl = this._lockFile.descriptionLocation;
-        this.includeFilters = this._lockFile.includePatterns;
-        this.excludeFilters = this._lockFile.excludePatterns;
-        const settings = this.settingsGetter();
-        await this.loadNodes(settings.clearCache);
-        if (this.rawRootNode) {
-            this.refreshView();
+    public async loadLockFile(path: string, clientOrPluginName?: string): Promise<void> {
```

Comment:
since we're not using lock files anymore, we should probably rename that method and the associated fields for consistency

Reply:
Oh yeah! I'll make the update